orderbook/src/account_balances.rs

```diff
@@ -143,6 +152,28 @@ impl BalanceFetching for Web3BalanceFetcher {
         };
         let _ = self._register_many(subscriptions.into_iter()).await;
     }
+
+    async fn can_transfer(&self, token: H160, from: H160, amount: U256) -> bool {
+        let instance = ERC20::at(&self.web3, token);
+        let calldata = instance
+            .transfer_from(from, self.settlement_contract, amount)
+            .tx
+            .data
+            .unwrap();
+        let call_request = CallRequest {
+            from: Some(self.allowance_manager),
+            to: Some(token),
+            data: Some(calldata),
+            ..Default::default()
+        };
+        let block = Some(BlockId::Number(BlockNumber::Latest));
+        const TRUE: [u8; 32] =
+            hex!("0000000000000000000000000000000000000000000000000000000000000001");
+        match self.web3.eth().call(call_request, block).await {
+            Ok(bytes) => bytes.0 == TRUE,
```

Comment:
I see. This is the same thing handleod in https://github.com/gnosis/gp-v2-contracts/blob/26e5eb19e639bc62f50c604a1dd38348799e520c/src/contracts/libraries/GPv2SafeERC20.sol#L110 . I wonder if we can call this same library function from here to get exactly the same behavior.

Reply:
I think we can't because it's an internal library (no public function).

Reply:
@fleupold What about this thread? Merge even though we would fail on those tokens? Not sure how many tokens are still affected now that it's 2 years after that blog post.
If we wanted to handle this better I'd have to go back to using rust-web3 directly without ethcontract and manually inspect the response bytes.

Reply:
Right, I forgot about this! How does ethcontract handle a contract that is returning no bytes when it expects a boolean (cc @nlordell)? The most known token to me in that list is OMG (which still has reasonable liquidity so we could test with it). 

Maybe we could actually go back to the manual call implementation and consider a 0 length return data a success?

Reply:
From having touched the token decoding code recently I believe ethcontract would return this as a decoding error. I think your last suggestion makes sense.

Reply:
> From having touched the token decoding code recently I believe ethcontract would return this as a decoding error

That is my assumption as well.

Reply:
I think, what would make the most sense here would be to do a raw `eth_call`:
```rust
let response = web3.eth().call(
    CallRequest {
                from: Some(owner_address),
                to: Some(token_address),
                data: token.transfer(amount).tx.data,
    },
    Some(block_id),
).await?;
ensure!(is_empty_bytes_or_true(response), "transfer failed"); 
```

Reply:
Changed it back to this version.